### PR TITLE
feat: tab visibility in vault

### PR DESCRIPTION
**Pull request by djaiss**
This PR lets you control which tab should be visible in a vault.

image

### CODE REVIEW

File: resources/js/Pages/Vault/Settings/Partials/TabVisibility.vue [hunk 1]
```diff
@@ -0,0 +1,132 @@
...
+
+onMounted(() => {
+  form.show_group_tab = props.data.visibility.show_group_tab;
+  form.show_tasks_tab = props.data.visibility.show_tasks_tab;
+  form.show_files_tab = props.data.visibility.show_files_tab;
+  form.show_journal_tab = props.data.visibility.show_journal_tab;
+});
+
+const update = () => {
+  axios.put(props.data.url.update_tab_visibility, form);
```
Comment (asbiin) [inline resources/js/Pages/Vault/Settings/Partials/TabVisibility.vue]: when using a `useForm`, it makes more sense to use it to send the request, like
```suggestion
  form.put(props.data.url.update_tab_visibility);
```
(but the response must be an Inertia Request ...)

File: app/Domains/Vault/ManageVaultSettings/Web/Controllers/VaultSettingsTabVisibilityController.php [hunk 2]
```diff
@@ -0,0 +1,26 @@
...
+    public function update(Request $request, int $vaultId)
+    {
+        $data = [
+            'account_id' => Auth::user()->account_id,
+            'author_id' => Auth::id(),
+            'vault_id' => $vaultId,
+            'show_group_tab' => $request->input('show_group_tab'),
+            'show_tasks_tab' => $request->input('show_tasks_tab'),
+            'show_files_tab' => $request->input('show_files_tab'),
+            'show_journal_tab' => $request->input('show_journal_tab'),
```
Comment (asbiin) [inline app/Domains/Vault/ManageVaultSettings/Web/Controllers/VaultSettingsTabVisibilityController.php]: You can use:
```suggestion
            'show_group_tab' => $request->boolean('show_group_tab'),
            'show_tasks_tab' => $request->boolean('show_tasks_tab'),
            'show_files_tab' => $request->boolean('show_files_tab'),
            'show_journal_tab' => $request->boolean('show_journal_tab'),
```

**Review by asbiin — APPROVED** [review]
Nice work!
See my consistency changes:

File: resources/js/Pages/Vault/Settings/Partials/TabVisibility.vue (hunk 1 above)
Comment (djaiss) [reply resources/js/Pages/Vault/Settings/Partials/TabVisibility.vue]: fixed

**Pull request merged**